### PR TITLE
feat(explorer): support Sepolia network

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

Added Sepolia network support in Explorer with minimum changes.

Charts and tokens table on main page are hidden for Sepolia, since there is no Subgraph deployment and we don't plan to deploy it.

  # To Test

Full Explorer regression test

### CODE REVIEW

File: apps/explorer/src/cowSdk.ts [hunk 1]
```diff
@@ -4,17 +4,20 @@ import { MetadataApi } from '@cowprotocol/app-data'
...
+  const [mainnetUrl, gcUrl, goerliUrl, sepoliaUrl] = [
     process.env.REACT_APP_SUBGRAPH_URL_MAINNET || undefined,
     process.env.REACT_APP_SUBGRAPH_URL_GNOSIS_CHAIN || undefined,
     process.env.REACT_APP_SUBGRAPH_URL_GOERLI || undefined,
+    process.env.REACT_APP_SUBGRAPH_URL_SEPOLIA || undefined,
   ]
 
   return {
     ...SUBGRAPH_PROD_CONFIG,
+    ...{ [SupportedChainId.SEPOLIA]: 'https://api.thegraph.com/subgraphs/name/cowprotocol/sepolia' },
```
[the remark this answers is not in the document]
Comment (shoom3301) [reply apps/explorer/src/cowSdk.ts]: Remove it

File: apps/explorer/src/api/tenderly/tenderlyApi.ts [hunk 2]
```diff
@@ -14,59 +13,58 @@ import {
...
-}
-
-function _getApiBaseUrl(networkId: Network): string {
+function _getApiBaseUrl(networkId: SupportedChainId): string {
   const baseUrl = API_BASE_URLs[networkId]
 
   if (!baseUrl) {
-    throw new Error('Unsupported Network. The tenderly API is not available in the Network ' + networkId)
+    throw new Error(
+      'Unsupported SupportedChainId. The tenderly API is not available in the SupportedChainId ' + networkId
```
Comment (alfetopito) [inline apps/explorer/src/api/tenderly/tenderlyApi.ts:30]: Here we could probably keep `Network`.

File: apps/explorer/src/api/tenderly/tenderlyApi.ts [hunk 3]
```diff
@@ -14,59 +13,58 @@ import {
...
+import { mapSupportedNetworks, SupportedChainId } from '@cowprotocol/cow-sdk'
 
 export const ALIAS_TRADER_NAME = 'Trader'
 const COW_PROTOCOL_CONTRACT_NAME = 'GPv2Settlement'
-const API_BASE_URLs = _urlAvailableNetwork()
 
-function _urlAvailableNetwork(): Partial<Record<Network, string>> {
-  const urlNetwork = (_networkId: Network): string => `${TENDERLY_API_URL}/${_networkId}`
+const API_BASE_URLs: Record<SupportedChainId, string> = mapSupportedNetworks(
+  (_networkId: SupportedChainId): string => `${TENDERLY_API_URL}/${_networkId}`
```
Comment (alfetopito) [inline apps/explorer/src/api/tenderly/tenderlyApi.ts:22]: Nitpick: why the `_` prefix?

File: apps/explorer/src/explorer/ExplorerApp.tsx [hunk 4]
```diff
@@ -143,8 +143,6 @@ export const ExplorerApp: React.FC = () => {
         <Router basename={process.env.BASE_URL}>
           <StateUpdaters />
           <Routes>
-            <Route path="/mainnet" element={<RedirectMainnet />} />
-            <Route path="/xdai" element={<RedirectXdai />} />
```
Comment (alfetopito) [inline apps/explorer/src/explorer/ExplorerApp.tsx:147]: Can you keep these?
It's there for historical reasons in case an old url is accessed.

**Review by alfetopito — COMMENTED** [review]
Everything is working great!

Just a few comments

File: apps/explorer/src/api/tenderly/tenderlyApi.ts (hunk 2 above)
Comment (shoom3301) [reply apps/explorer/src/api/tenderly/tenderlyApi.ts:30]: Fixed

File: apps/explorer/src/api/tenderly/tenderlyApi.ts (hunk 3 above)
Comment (shoom3301) [reply apps/explorer/src/api/tenderly/tenderlyApi.ts:22]: image

I have no idea :) Some other developer added it

File: apps/explorer/src/explorer/ExplorerApp.tsx (hunk 4 above)
Comment (shoom3301) [reply apps/explorer/src/explorer/ExplorerApp.tsx:147]: Do you think it is still relevant?
I've joined the team 1.5 years ago and have never seen urls with this prefixes.
Looks like it's super outdated

**Review by anxolin — COMMENTED** [review]
@shoom3301 or @fairlighteth could you style the searchbox, so its really centered when we don't have the token table?

image

File: apps/explorer/src/api/tenderly/tenderlyApi.ts (hunk 3 above)
Comment (alfetopito) [reply apps/explorer/src/api/tenderly/tenderlyApi.ts:22]: Oh ok. Mind updating it?

File: apps/explorer/src/explorer/ExplorerApp.tsx (hunk 4 above)
Comment (alfetopito) [reply apps/explorer/src/explorer/ExplorerApp.tsx:147]: Well, we were still creating those 😓 

The last one I could find is from ~1y ago:

https://cowservices.slack.com/archives/C03S445V822/p1673720749446989

**Comment by shoom3301** [comment]
image

@anxolin like this?

File: apps/explorer/src/explorer/ExplorerApp.tsx (hunk 4 above)
Comment (shoom3301) [reply apps/explorer/src/explorer/ExplorerApp.tsx]: Ok ok, added it back

File: apps/explorer/src/api/tenderly/tenderlyApi.ts (hunk 3 above)
Comment (shoom3301) [reply apps/explorer/src/api/tenderly/tenderlyApi.ts:22]: image

There are many variables with the prefix, let's do the refactor in another PR?

File: apps/explorer/src/consts/network.ts [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  name: string
+  url: string
+}
+
+export const CHAIN_ID_TO_NAME: Record<SupportedChainId, string> = {
+  [SupportedChainId.MAINNET]: 'Ethereum',
+  [SupportedChainId.GNOSIS_CHAIN]: 'Gnosis Chain',
+  [SupportedChainId.GOERLI]: 'Görli',
+  [SupportedChainId.SEPOLIA]: 'Sepolia',
+}
```
Comment (anxolin) [inline apps/explorer/src/consts/network.ts:14]: isn't this a duplication?

**Review by anxolin — APPROVED** [review]
Works great!

It feels we still need to continue to cleanup. The explorer had smaller standards than CoW Swap, and there's definitely duplication. 

Not in a rush to remove duplication, but feels like good moment to continue to clean

File: apps/explorer/src/consts/network.ts (hunk 5 above)
Comment (shoom3301) [reply apps/explorer/src/consts/network.ts:14]: I can't find other similar configs

**Comment by shoom3301** [comment]
> Works great!
> 
> It feels we still need to continue to cleanup. The explorer had smaller standards than CoW Swap, and there's definitely duplication.
> 
> Not in a rush to remove duplication, but feels like good moment to continue to clean

Yes, Explorer definitely needs refactoring. I will work on it just after merging this PR

**Comment by elena-zh** [comment]
Missing feature: when search for an [address on Explorer in a chain where no orders](https://explorer-git-feat-explorer-sepolia-1-cowswap.vercel.app/goerli/address/0x8FAb71C0d4272698A3B2d1F3Ed5FC3c1B9b3E531), it shows info about orders in another chains (if any) 
image

See an example for this [address](https://explorer-git-feat-explorer-sepolia-1-cowswap.vercel.app/sepolia/address/0x9FA3c00a92Ec5f96B1Ad2527ab41B3932EFEDa58): 
image

File: apps/explorer/src/api/tenderly/tenderlyApi.ts (hunk 3 above)
Comment (alfetopito) [reply apps/explorer/src/api/tenderly/tenderlyApi.ts:22]: Ok

File: apps/explorer/src/consts/subgraphUrls.ts [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import { SUBGRAPH_PROD_CONFIG, SupportedChainId } from '@cowprotocol/cow-sdk'
+
+function getSubgraphUrl(chainId: SupportedChainId, suffix: string): string | null {
+  return process.env[`REACT_APP_SUBGRAPH_URL_${suffix}`] || SUBGRAPH_PROD_CONFIG[chainId] || null
+}
+
+/**
+ * When value is undefined, then Subgraph is not supported for the network
```
Comment (alfetopito) [inline apps/explorer/src/consts/subgraphUrls.ts:8]:
```suggestion
 * When value is null, then Subgraph is not supported for the network
```

**Review by elena-zh — APPROVED** [review]
Looks good to me now!

**Pull request merged**